flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/table/MySqlTableSourceFactoryTest.java

```diff
@@ -190,6 +194,30 @@ public void testEnableParallelReadSourceLatestOffset() {
         assertEquals(expectedSource, actualSource);
     }
 
+    @Test
+    public void testOptionalOptions() {
+        Factory factory = new MySqlTableSourceFactory();
+        Set<ConfigOption<?>> optionalOptions = factory.optionalOptions();
+        for (Field field : MySqlSourceOptions.class.getDeclaredFields()) {
+            if (Modifier.isStatic(field.getModifiers()) && field.getType() == ConfigOption.class) {
+                try {
+                    field.setAccessible(true);
+                    ConfigOption<?> option = (ConfigOption<?>) field.get(factory);
+                    if (option.hasDefaultValue()) {
```

Comment:
Although the test works well now,  IIUC there's no rule that the optional option must contains a default value, thus this test may be problematic after we add an optional option without default value.
I suggest we can simple add unit test in `testOptionalProperties`, WDYT ?

Reply:
`there's no rule that the optional option must contains a default value, thus this test may be problematic after we add an optional option without default value.`
That's right, but what about the rule that options have a default value must be optional options?  

The test will just catch options have a default value but not in the `optionalOptions`. 

And I do agree to add unit test for `slot.name` in `testOptionalProperties` for consistent.

Reply:
>That's right, but what about the rule that options have a default value must be optional options?

I think no this kind of rule in Flink or Flink CDC

>The test will just catch options have a default value but not in the optionalOptions.Yes, I agree that 
Yes, you're right. Thus we cannot check the new add optional option without default value.

Reply:
>That's right, but what about the rule that options have a default value must be optional options?

I think no this kind of rule in Flink or Flink CDC
>The test will just catch options have a default value but not in the optionalOptions.Yes, I agree that 

Yes, you're right. Thus we cannot check the new add optional option without default value.

Reply:
Could we delete the test `testOptionalOptions` as we discussed above? @dongdongking008

Reply:
ok.